syntax="proto3";

package ImageStore;

// The ImageStore service definition.
service is {

    //**********ImageStore Interfaces***************
    // Read external interface
    rpc Read (ReadReq) returns (stream ReadResp) {}

    // Store external interface
    rpc Store (stream StoreReq) returns (StoreResp) {}

    // Remove external interface
    rpc Remove (RemoveReq) returns (RemoveResp) {}

}

// The request message for Read interface
message ReadReq {
    string readKeyname = 1;
}

// Response message for Read interface
message ReadResp {
    bytes chunk = 1;
}

// The request message for Store interface
message StoreReq {
    bytes chunk = 1;
    string memoryType = 2;
}

// The response message for Store interface
message StoreResp {
    string storeKeyname = 1;
}

// The request message for GetConfig
message RemoveReq {
    string remKeyname = 1;
}

// The response message for GetConfig
message RemoveResp {
   
}
